superset/views/datasource.py

```diff
@@ -0,0 +1,49 @@
+# -*- coding: utf-8 -*-
+# pylint: disable=C,R,W
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import json
+
+from flask import request
+from flask_appbuilder import expose
+from flask_babel import gettext as __
+
+from superset import appbuilder, db
+from superset.connectors.connector_registry import ConnectorRegistry
+from .base import BaseSupersetView, check_ownership, json_error_response
+
+
+class Datasource(BaseSupersetView):
+    """Datasource-related views"""
+    @expose('/save/', methods=['POST'])
+    def save(self):
+        datasource = json.loads(request.form.get('data'))
+        datasource_id = datasource.get('id')
+        datasource_type = datasource.get('type')
+        orm_datasource = ConnectorRegistry.get_datasource(
+            datasource_type, datasource_id, db.session)
+
+        if not check_ownership(orm_datasource, raise_if_false=False):
+            return json_error_response(
+                __(
+                    'You are not authorized to modify '
+                    'this data source configuration'),
+                status='401',
```

Comment:
hey @mistercrunch in the new datasource editor you need to be an owner to edit a datasource, previously anyone could edit datasources (this is still the case in the crud view). Allowing anyone to edit is useful because many people may want to add columns or metrics to a datasource. Do you think it's necessary to only have owners edit a datasource? 

I can change this to make it configurable, but just wanted some more info from you about this change.

Reply:
Hmmm, how about creating a new permission `can_edit_any_datasource` ?

Reply:
That sounds good, I'll take a look.